src/main/java/io/nats/client/impl/NatsJetStreamSubscription.java

```diff
@@ -150,6 +151,14 @@ public void pull(int batchSize) {
         throw new IllegalStateException(SUBSCRIPTION_TYPE_DOES_NOT_SUPPORT_PULL);
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void pull(PullRequestOptions pullRequestOptions) {
+        throw new IllegalStateException(SUBSCRIPTION_TYPE_DOES_NOT_SUPPORT_PULL);
+    }
```

Comment:
Only pull subs allow this call and this is the base class, not the pull version so is the default behavior. For simplification, the interface design should be more like .NET where the pull interface is not part of the JS Sub interface